nodes/exchange/FCStd_read.py

```diff
@@ -0,0 +1,103 @@
+from sverchok.dependencies import FreeCAD
+from sverchok.utils.dummy_nodes import add_dummy
+
+if FreeCAD is None:
+    add_dummy('SvReadFCStdNode', 'SvReadFCStdNode', 'FreeCAD')
+
+else:
+    F = FreeCAD
+    import bpy,sys
+    from bpy.props import IntProperty, FloatProperty, StringProperty, BoolProperty
+    from sverchok.node_tree import SverchCustomTreeNode
+    from numpy import ndarray
+
+    class SvReadFCStdNode(bpy.types.Node, SverchCustomTreeNode):
+        ''' SvReadFCStdNode '''
+        bl_idname = 'SvReadFCStdNode'
+        bl_label = 'Read FCStd'
+        bl_icon = 'IMPORT'
+        solid_catergory = "Inputs"
+        
+        read_update : BoolProperty(name="read_update", default=True)
+        part_filter : StringProperty(name="part_filter", default="", description="use ',' to separate name with no space: part1,part2,... ")
+        inv_filter : BoolProperty(name="inv_filter", default=False)
+
+        def draw_buttons(self, context, layout):
+
+            layout.label(text="filter by name:")
+
+            col = layout.column(align=True)
+            col.prop(self, 'part_filter',text="")
+            col.prop(self, 'inv_filter')
+            col.prop(self, 'read_update')          
+
+        def sv_init(self, context):
+            self.inputs.new('SvFilePathSocket', "File Path")       
+            self.outputs.new('SvSolidSocket', "Solid")
+
+        def process(self):
+            if not any(socket.is_linked for socket in self.outputs):
+                return
+            if not self.inputs['File Path'].is_linked:
+                return            
+
+            if self.read_update:
+                
+                files = self.inputs['File Path'].sv_get()[0]
+                solids = []
+
+                for f in files:
+                    S = LoadSolid(f,self.part_filter,self.inv_filter)
+                    for s in S:
+                        solids.append(s)
+                
+                self.outputs['Solid'].sv_set(solids)
+            
+            else:
+                return
+
+
+def LoadSolid(fc_file,part_filter,inv_filter):
+    solids = []
+
+    if ',' in part_filter:
+        part_filter = part_filter.split(',')
+
+    elif len(part_filter)==0:
+        part_filter = ['all']

```

Comment:
It is not too good to use one variable for two different things. I mean `part_filter` should be either string ot list.

Reply:
Yes, I'm trying another solution adding a filter input . 
I didn't find any string input node to use so I scripted a simple one to make a test.
It works better and it's a lot more consistent also if require a lot of nodes if one need to import a lot of parts, indeed it could be added in the future a specific node to read parts label and pass the wanted selection.

![filter_input](https://user-images.githubusercontent.com/8501792/97336074-e319d900-187e-11eb-943b-be1b1d8789d9.png)